sdk/core/azure-core/inc/azure/core/nullable.hpp

```diff
@@ -261,10 +261,72 @@ namespace Azure { namespace Core {
       return std::move(m_value);
     }
 
+    // observers
+
     /**
      * @brief `operator bool` on the condition of #Azure::Core::Nullable::HasValue.
      */
-    explicit operator bool() const noexcept { return HasValue(); }
+    constexpr explicit operator bool() const noexcept { return HasValue(); }
+
+    /**
+     * @brief Accesses the contained value.
+     * @return Returns a pointer to the contained value
+     * @warning The behavior is undefined if `*this` does not contain a value.
+     * @note This operator does not check whether the optional contains a value!
+             You can do so manually by using #HasValue() or simply operator #bool().
+             Alternatively, if checked access is needed, #GetValue() or #ValueOr() may be used.
+     */
+    constexpr const T* operator->() const { return std::addressof(m_value); }
+
+    /**
+     * @brief Accesses the contained value.
+     * @return Returns a pointer to the contained value
+     * @warning The behavior is undefined if `*this` does not contain a value.
+     * @note This operator does not check whether the optional contains a value!
+             You can do so manually by using #HasValue() or simply operator #bool().
+             Alternatively, if checked access is needed, #GetValue() or #ValueOr() may be used.
+     */
+    constexpr T* operator->() { return _STD addressof(m_value); }
+
+    /**
+     * @brief Accesses the contained value.
+     * @return Returns a reference to the contained value
+     * @warning The behavior is undefined if `*this` does not contain a value.
+     * @note This operator does not check whether the optional contains a value!
+             You can do so manually by using #HasValue() or simply operator #bool().
+             Alternatively, if checked access is needed, #GetValue() or #ValueOr() may be used.
+     */
+    constexpr const T& operator*() const& { return m_value; }
```

Comment:
What's the difference between the `*` operator and the `->` operator? Both APIs briefs say the same thing.
As the  caller, when should I use one versus the other? Maybe clarify this a bit more in the doc comments.

Reply:
`x->` is an alias for `(*x).`
It was added to simplify something like `(*pointer).Foo()`  to just `pointer->Foo()`.
The compiler will replace the `x->` for the `(*x).` equivalence.
The same thing happen with the references,  it becomes an alias for a pointer `p->`
So, something like
```cpp
int Foo(std::string& ref) {
   ref.append("Hi");
 }
```
gets translated by compiler to
```cpp
int Foo(std::string* ref) {
   ref->append("Hi");  // later to  (*ref).append("Hi");
 }
```

Since it could become personal preference which one to use, we should always offer both operators and never just one or the other.  (Diversity and inclusion <3 )

Reply:
They are used to access members of a type.  Which one you use depends on the situation and sometimes your style patterns.  For example there are a lot places where you can interchange `x->y` for `(*x).y` depending on your style patterns.